provider/boltmem/boltmem.go

```diff
@@ -261,15 +261,15 @@ func (s *Silences) Mutes(lset model.LabelSet) bool {
 
 // All returns all existing silences.
 func (s *Silences) All() (*types.SilencesQueryResponse, error) {
-	return s.Query(len(s.k), 0)
+	return s.Query(len(s.sils), 0, types.ByCreatedAt)
```

Comment:
That doesn't solve the problem either.

I'm personally inclined to just remove `All()` from the interface, since if a user is REALLY wanting all the silences, they can query the listSilences endpoint with some arbitrarily large `limit` value.